### ISSUE TITLE
Bug: Improper parsing of ansi codes

### ISSUE THREAD

**Issue by zevaryx** [FHPythonUtils/AnsiToImg#3]
## Bug

- [x] I have read the comment above and have completed each step
- [x] I have filled out the system info
- [x] I have described the bug, filled in the expected outcome and the actual
outcome including screenshots where appropriate

### System info

- OS: Windows 11 build 25115
- Browser: N/A
- Version: 2022

### Describe the bug
Parsing the following test results in `ValueError: invalid literal for int() with base 10: '3;96'`
```ansi
\x1b[34m╭─\x1b[0m\x1b[34m────────────────────────────────────────────────────────────────────── \x1b[0m\x1b[1;34m<\x1b[0m\x1b[1;95mclass\x1b[0m\x1b[39m \x1b[0m\x1b[32m'jarvis.config.JarvisConfig'\x1b[0m\x1b[1;34m>\x1b[0m\x1b[34m ──────────────────────────────────────────────────────────────────────\x1b[0m\x1b[34m─╮\x1b[0m
\x1b[34m│\x1b[0m \x1b[3;96mclass \x1b[0m\x1b[1;31mJarvisConfig\x1b[0m\x1b[1m(\x1b[0m*args: list, **kwargs: dict\x1b[1m)\x1b[0m:
                                                                                         \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m
                                                   \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m  \x1b[3;33mOPTIONAL\x1b[0m = \x1b[1m{\x1b[0m\x1b[32m'sync'\x1b[0m: \x1b[3;91mFalse\x1b[0m, \x1b[32m'log_level'\x1b[0m: \x1b[32m'WARNING'\x1b[0m, \x1b[32m'cogs'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'events'\x1b[0m: \x1b[3;92mTrue\x1b[0m, \x1b[32m'gitlab_token'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'max_messages'\x1b[0m: \x1b[1;36m1000\x1b[0m, \x1b[32m'twitter'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'reddit'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'rook_token'\x1b[0m: \x1b[3;35mNone\x1b[0m\x1b[1m}\x1b[0m \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m  \x1b[3;33mREQUIRED\x1b[0m = \x1b[1m(\x1b[0m\x1b[32m'token'\x1b[0m, \x1b[32m'mongo'\x1b[0m, \x1b[32m'urls'\x1b[0m, \x1b[32m'redis'\x1b[0m\x1b[1m)\x1b[0m                                                                                                                                    \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m \x1b[3;33mfrom_yaml\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31mfrom_yaml\x1b[0m\x1b[1m(\x1b[0mfilepath: Union\x1b[1m[\x1b[0mpathlib.Path, str\x1b[1m]\x1b[0m = \x1b[1;35mWindowsPath\x1b[0m\x1b[1m(\x1b[0m\x1b[32m'config.yaml'\x1b[0m\x1b[1m)\x1b[0m\x1b[1m)\x1b[0m -> \x1b[32m'Config'\x1b[0m: \x1b[2mLoad the yaml config file.\x1b[0m                                                 \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m      \x1b[3;33minit\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31minit\x1b[0m\x1b[1m(\x1b[0mself, **kwargs: dict\x1b[1m)\x1b[0m -> \x1b[3;35mNone\x1b[0m: \x1b[2mInitialize the object.\x1b[0m                                                                                                         \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m    \x1b[3;33mreload\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31mreload\x1b[0m\x1b[1m(\x1b[0m\x1b[1m)\x1b[0m -> bool: \x1b[2mReload the config.\x1b[0m                                                                                                                               \x1b[34m│\x1b[0m
\x1b[34m╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯\x1b[0m
```

The error occurs in `ansitoimg.utils` on line 81. I think this is due to incomplete parsing of ansi modifiers like bold, italic, etc., or at least improper checking for these

### Expected outcome
A proper render of the text, similar to the following:
![image](https://user-images.githubusercontent.com/57541873/169113589-e7c9b9b3-9f50-482d-b02a-7d80f89b6942.png)

### Actual outcome
The aforementioned error

**Comment by FredHappyface** [comment]
Thank you for opening the issue.

This should be fixed in the latest version (2022) 

```
vagrant@ubuntu2004:~$ pip3 install base24builder
Collecting base24builder
  Downloading base24builder-2022-py3-none-any.whl (15 kB)
Collecting pystache<2,>=0.6.0
  Downloading pystache-0.6.0.tar.gz (78 kB)
     |████████████████████████████████| 78 kB 3.0 MB/s
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting aiofiles<2,>=0.8.0
  Downloading aiofiles-0.8.0-py3-none-any.whl (13 kB)
Collecting PyYAML<7,>=6.0
  Downloading PyYAML-6.0-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (701 kB)
     |████████████████████████████████| 701 kB 10.9 MB/s
Building wheels for collected packages: pystache
  Building wheel for pystache (PEP 517) ... done
  Created wheel for pystache: filename=pystache-0.6.0-py3-none-any.whl size=83618 sha256=2f28e6d2fdaefe97bfa703bb057b77e302ebd132009470074f845a714a97d08a
  Stored in directory: /home/vagrant/.cache/pip/wheels/b1/d4/cc/b2293fa560ba9e1c5724843437688e20235fcbdc211cb37e37
Successfully built pystache
Installing collected packages: pystache, aiofiles, PyYAML, base24builder
  WARNING: The scripts pystache and pystache-test are installed in '/home/vagrant/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script base24 is installed in '/home/vagrant/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed PyYAML-6.0 aiofiles-0.8.0 base24builder-2022 pystache-0.6.0
```

**Comment by FredHappyface** [comment]
I've tested with the rich save_svg function. Can you confirm this is more inline with what you are expecting? Going forward it looks like this may be a more capable alternative to this library. (https://rich.readthedocs.io/en/latest/console.html#exporting). What I may do going forward is stand on the shoulders of giants and use this under the hood to generate images 

```py
raw = """
\x1b[34m╭─\x1b[0m\x1b[34m────────────────────────────────────────────────────────────────────── \x1b[0m\x1b[1;34m<\x1b[0m\x1b[1;95mclass\x1b[0m\x1b[39m \x1b[0m\x1b[32m'jarvis.config.JarvisConfig'\x1b[0m\x1b[1;34m>\x1b[0m\x1b[34m ──────────────────────────────────────────────────────────────────────\x1b[0m\x1b[34m─╮\x1b[0m
\x1b[34m│\x1b[0m \x1b[3;96mclass \x1b[0m\x1b[1;31mJarvisConfig\x1b[0m\x1b[1m(\x1b[0m*args: list, **kwargs: dict\x1b[1m)\x1b[0m:
                                                                                         \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m
                                                   \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m  \x1b[3;33mOPTIONAL\x1b[0m = \x1b[1m{\x1b[0m\x1b[32m'sync'\x1b[0m: \x1b[3;91mFalse\x1b[0m, \x1b[32m'log_level'\x1b[0m: \x1b[32m'WARNING'\x1b[0m, \x1b[32m'cogs'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'events'\x1b[0m: \x1b[3;92mTrue\x1b[0m, \x1b[32m'gitlab_token'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'max_messages'\x1b[0m: \x1b[1;36m1000\x1b[0m, \x1b[32m'twitter'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'reddit'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'rook_token'\x1b[0m: \x1b[3;35mNone\x1b[0m\x1b[1m}\x1b[0m \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m  \x1b[3;33mREQUIRED\x1b[0m = \x1b[1m(\x1b[0m\x1b[32m'token'\x1b[0m, \x1b[32m'mongo'\x1b[0m, \x1b[32m'urls'\x1b[0m, \x1b[32m'redis'\x1b[0m\x1b[1m)\x1b[0m                                                                                                                                    \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m \x1b[3;33mfrom_yaml\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31mfrom_yaml\x1b[0m\x1b[1m(\x1b[0mfilepath: Union\x1b[1m[\x1b[0mpathlib.Path, str\x1b[1m]\x1b[0m = \x1b[1;35mWindowsPath\x1b[0m\x1b[1m(\x1b[0m\x1b[32m'config.yaml'\x1b[0m\x1b[1m)\x1b[0m\x1b[1m)\x1b[0m -> \x1b[32m'Config'\x1b[0m: \x1b[2mLoad the yaml config file.\x1b[0m                                                 \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m      \x1b[3;33minit\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31minit\x1b[0m\x1b[1m(\x1b[0mself, **kwargs: dict\x1b[1m)\x1b[0m -> \x1b[3;35mNone\x1b[0m: \x1b[2mInitialize the object.\x1b[0m                                                                                                         \x1b[34m│\x1b[0m
\x1b[34m│\x1b[0m    \x1b[3;33mreload\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31mreload\x1b[0m\x1b[1m(\x1b[0m\x1b[1m)\x1b[0m -> bool: \x1b[2mReload the config.\x1b[0m                                                                                                                               \x1b[34m│\x1b[0m
\x1b[34m╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯\x1b[0m
"""


from rich.console import Console
from rich import text

console = Console(width=200, record=True)

ansi = text.Text.from_ansi(raw)
console.print(ansi)
console.save_svg("test.svg")


```



![image](https://user-images.githubusercontent.com/41634689/170381648-8482b651-279b-47d3-9339-3bdfba289e1a.png)

**Comment by FredHappyface** [comment]
Fixed in 2022.1

**Comment by zevaryx** [comment]
> I've tested with the rich save_svg function. Can you confirm this is more inline with what you are expecting? Going forward it looks like this may be a more capable alternative to this library. (https://rich.readthedocs.io/en/latest/console.html#exporting). What I may do going forward is stand on the shoulders of giants and use this under the hood to generate images
> 
> ```python
> raw = """
> \x1b[34m╭─\x1b[0m\x1b[34m────────────────────────────────────────────────────────────────────── \x1b[0m\x1b[1;34m<\x1b[0m\x1b[1;95mclass\x1b[0m\x1b[39m \x1b[0m\x1b[32m'jarvis.config.JarvisConfig'\x1b[0m\x1b[1;34m>\x1b[0m\x1b[34m ──────────────────────────────────────────────────────────────────────\x1b[0m\x1b[34m─╮\x1b[0m
> \x1b[34m│\x1b[0m \x1b[3;96mclass \x1b[0m\x1b[1;31mJarvisConfig\x1b[0m\x1b[1m(\x1b[0m*args: list, **kwargs: dict\x1b[1m)\x1b[0m:
>                                                                                          \x1b[34m│\x1b[0m
> \x1b[34m│\x1b[0m
>                                                    \x1b[34m│\x1b[0m
> \x1b[34m│\x1b[0m  \x1b[3;33mOPTIONAL\x1b[0m = \x1b[1m{\x1b[0m\x1b[32m'sync'\x1b[0m: \x1b[3;91mFalse\x1b[0m, \x1b[32m'log_level'\x1b[0m: \x1b[32m'WARNING'\x1b[0m, \x1b[32m'cogs'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'events'\x1b[0m: \x1b[3;92mTrue\x1b[0m, \x1b[32m'gitlab_token'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'max_messages'\x1b[0m: \x1b[1;36m1000\x1b[0m, \x1b[32m'twitter'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'reddit'\x1b[0m: \x1b[3;35mNone\x1b[0m, \x1b[32m'rook_token'\x1b[0m: \x1b[3;35mNone\x1b[0m\x1b[1m}\x1b[0m \x1b[34m│\x1b[0m
> \x1b[34m│\x1b[0m  \x1b[3;33mREQUIRED\x1b[0m = \x1b[1m(\x1b[0m\x1b[32m'token'\x1b[0m, \x1b[32m'mongo'\x1b[0m, \x1b[32m'urls'\x1b[0m, \x1b[32m'redis'\x1b[0m\x1b[1m)\x1b[0m                                                                                                                                    \x1b[34m│\x1b[0m
> \x1b[34m│\x1b[0m \x1b[3;33mfrom_yaml\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31mfrom_yaml\x1b[0m\x1b[1m(\x1b[0mfilepath: Union\x1b[1m[\x1b[0mpathlib.Path, str\x1b[1m]\x1b[0m = \x1b[1;35mWindowsPath\x1b[0m\x1b[1m(\x1b[0m\x1b[32m'config.yaml'\x1b[0m\x1b[1m)\x1b[0m\x1b[1m)\x1b[0m -> \x1b[32m'Config'\x1b[0m: \x1b[2mLoad the yaml config file.\x1b[0m                                                 \x1b[34m│\x1b[0m
> \x1b[34m│\x1b[0m      \x1b[3;33minit\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31minit\x1b[0m\x1b[1m(\x1b[0mself, **kwargs: dict\x1b[1m)\x1b[0m -> \x1b[3;35mNone\x1b[0m: \x1b[2mInitialize the object.\x1b[0m                                                                                                         \x1b[34m│\x1b[0m
> \x1b[34m│\x1b[0m    \x1b[3;33mreload\x1b[0m = \x1b[3;96mdef \x1b[0m\x1b[1;31mreload\x1b[0m\x1b[1m(\x1b[0m\x1b[1m)\x1b[0m -> bool: \x1b[2mReload the config.\x1b[0m                                                                                                                               \x1b[34m│\x1b[0m
> \x1b[34m╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯\x1b[0m
> """
> 
> 
> from rich.console import Console
> from rich import text
> 
> console = Console(width=200, record=True)
> 
> ansi = text.Text.from_ansi(raw)
> console.print(ansi)
> console.save_svg("test.svg")
> ```
> 
> ![image](https://github.com/assets/41634689/8482b651-279b-47d3-9339-3bdfba289e1a)

I just saw this, but this does work. I didn't even realize that Rich supported it, though I'd prefer to use your lib for various reasons (mainly png support)
